use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,491 @@
+```
+
+## Verify Intel QAT works in a Docker Kata Containers container
+
+The following instructions leverage an OpenSSL DockerFile that builds the 
+Intel QAT engine to allow OpenSSL to offload crypto functions. It is a 
+convenient way to test that VFIO device passthrough for the Intel QAT VF’s are
+working properly with the Kata Containers VM.
+
+## Build OpenSSL Intel QAT engine container
+
+There is an OpenSSL Intel QAT Dockerfile located at 
+[https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine](https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine). 
+Use this file to build a container image with an optimized OpenSSL engine for 
+Intel QAT. Using Docker build with the kata runtime can sometimes have issues. 
+Therefore, it is recommended to change the default runtime to runc before 
+doing a build. Instructions for this are below.
+
+__Note:__ The Intel QAT driver version in this container might not match the 
+Intel QAT driver compiled and loaded on the host.
+
+```
+$ cd $QAT_SRC
+$ curl -O $QAT_DOCKERFILE
+$ sudo sed -i 's/kata-runtime/runc/g' /etc/systemd/system/docker.service.d/50-runtime.conf
+$ sudo systemctl daemon-reload && sudo systemctl restart docker
+$ docker build -t openssl-qat-engine .
+```
+
+### Test Intel QAT in Docker
+
+The host should already be setup with 16 virtual functions of the Intel QAT 
+card bound to VFIO-PCI. This can be verified by looking in `/dev/VFIO` for a 
+listing of functions. Replace the number 90 with one of the VF’s exposed 
+in `/dev/VFIO`. It might be required to add an `IPC_LOCK` capability to your 
+Docker runtime depending on which rootfs is used. 
+
+```
+$ docker run -it --runtime=kata-runtime --cap-add=IPC_LOCK --device=/dev/VFIO/90 -v /dev:/dev -v ${QAT_CONF_LOCATION}:/etc openssl-qat-engine bash
+```
+
+Below are some commands to run in the container image to verify Intel QAT is 
+working
+
+```
+bash-5.0# cat /proc/modules
+bash-5.0# adf_ctl restart
+bash-5.0# adf_ctl status
+bash-5.0# openssl engine -c -t qat
+
+#Test with Intel QAT card acceleration
+bash-5.0# openssl speed -engine qat -elapsed -async_jobs 72 rsa2048 
+
+#Test with CPU acceleration
+bash-5.0# openssl speed -elapsed rsa2048
+```
+
+### Troubleshooting
+
+* Check that `/dev/VFIO` has VF’s enabled
+```
+$ ls /dev/VFIO
+57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  VFIO
+```
+* Check that the modules are loaded when inside the Kata Container
+```
+bash-5.0# cat /proc/modules
+qat_c62xvf 16384 - - Live 0x0000000000000000 (O)
+usdm_drv 86016 - - Live 0x0000000000000000 (O)
+intel_qat 184320 - - Live 0x0000000000000000 (O)
+```
+
+* Verify that at least the first `c6xxvf_dev0.conf` files is mounted inside the 
+container image in `/etc`. You will need one conf file for each VF passed into 
```

Comment:
We don't recognise "conf" as a word so please can you change to either "config" or "configuration".

Reply:
I will update.  As an FYI I don't recognize "recognise"  :)  Even my editor see's a red wavy line. LOL